File: gaia-sdk-javascript/src/graphql/request/type/DeletedSkillImpulse.ts

import {TenantKeyOne} from "./TenantKeyOne";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

/**
 * Impulse which indicates the result of a delete Skill impulse
 */
export class DeletedSkillImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "DeletedSkillImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:TenantKeyOne) => void) => this.push((registry) => {
        const entity = new TenantKeyOne();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/DeletedSkillImpulse.py

from gaia_sdk.graphql.request.type.TenantKeyOne import TenantKeyOne

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class DeletedSkillImpulse(list):
    """
    Impulse which indicates the result of a delete Skill impulse
    """

    def id(self):
        self.append(lambda x: "id")

    def data(self, config: Callable[['TenantKeyOne'], None]):
        def callback(registry: VariableRegistry):
            entity = TenantKeyOne()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
